src/scripts/archive_dataset/archive_dataset.py

```diff
@@ -0,0 +1,141 @@
+"""
+Archive a dataset and its derived datasets by copying them to the
+archive folder. A typical use case is to archive a dataset before a compatible
+schema change. Multiple compatible schema changes may happen to
+a single dataset, thus archives have an update number appended
+to their name. For example, if this is the third time we are
+archiving the `dataset_metadata_v2` dataset, the archive will be named
```

Comment:
Why is the version not increased instead when there is a schema change?

Reply:
Versions are increased in the case of an _incompatible_ schema change. This script is run during a compatible schema change, so no need to bump the version.